roles/kubernetes/node/templates/loadbalancer/haproxy.cfg.j2

```diff
@@ -21,7 +21,7 @@ defaults
 
 {% if loadbalancer_apiserver_healthcheck_port is defined -%}
 frontend healthz
-  bind *:{{ loadbalancer_apiserver_healthcheck_port }}
+  bind 127.0.0.1:{{ loadbalancer_apiserver_healthcheck_port }}
```

Comment:
The original issue say the IP address is not configurable.
In addition, this changes the IP address to the static 127.0.0.1 from 0.0.0.0.
I feel it would be nice to have `{{ loadbalancer_apiserver_healthcheck_address }}` with the default `0.0.0.0` and users can change the value if necessary for hardening environments to make this change effect small as possible with the default value.

Reply:
This LB is in place just so that control plane nodes can reach a `kube-apiserver` if the local `kube-apiserver` is offline for some reason, it is not meant to be used as a backend for an external LB solution so it fully makes sense to bind it to localhost.

My only concern here is IPv6 hardliners who will complain that 127.0.0.1 is IPv4 only and breaks their snowflake IPv6 only configuration.

@yankay can you find a solution that would work for IPv6-only setups?

Reply:
> This LB is in place just so that control plane nodes can reach a `kube-apiserver` if the local `kube-apiserver` is offline for some reason, it is not meant to be used as a backend for an external LB solution so it fully makes sense to bind it to localhost.
> 
> My only concern here is IPv6 hardliners who will complain that 127.0.0.1 is IPv4 only and breaks their snowflake IPv6 only configuration.
> 
> @yankay can you find a solution that would work for IPv6-only setups?

Thanks @cristicalin for the PR review.

The PR has been changed to support the dual_stack environment.
It's a little diffcult to support the IPv6 only environment.